### DISCUSSION TITLE
anyway to use astro?

### DISCUSSION THREAD

**Discussion by pipinosuke** [rehypejs/rehype#142] [Q&A] [answered]
i cant see good way to use this in my astro project

**Comment by ChristianMurphy** [comment]
Astro is built on rehype.
You can find out more on how to use plugins in the Astro documentation https://docs.astro.build/en/guides/markdown-content/#markdown-plugins

**Comment by pipinosuke** [comment]
yup, thats right
However, i cant import correctly `MyComponent` and `React.createElement`  in `astro.config.mjs` file.like below.
So, i am confused.

```mjs
    rehypePlugins: [
      [rehypeReact, {
        createElement: React.createElement,  //error
        components: {
          a: MyComponent, // error
        }}]
    ],
```

**Reply by Murderlon** [reply]
For questions specifically on Astro's abstraction of rehype, it's probably better to open a discussion there.

**Reply by ChristianMurphy** [reply, marked as answer]
Astro provides it's own renderer for rehype, you may not be able to use `rehype-react`, but the Astro team would be able to confirm.
Community support and discussions for Astro happen on their official discord https://github.com/withastro/astro#support

**Comment by pipinosuke** [comment]
exactly.
thank you two for help !!

**Comment by pipinosuke** [comment]
ぽにょ
